Submitty/Submitty
# [Feature:System] Student Override Back Into Course

### What is the current behavior?
When student's are dropped from a course by automated scripts, they have no way of rejoining the course without contacting the instructor. This can cause them to get behind on work, and the instructor has to manage readding students.


### What is the new behavior?
The No Access page for a course now has an option for students to rejoin a course, provided they meet certain conditions:

- The student was not manually removed from the course by the instructor
- The student was in the course to begin with
- It is within two weeks of the start of the term OR the student has accessed a gradeable, poll, forum post, course material, etc within the last 72 hours.

### Relationship to Other PRs and Issues
This PR address issue #9738 but does not close it, as future PRs will address other parts of that issue. This PR is followed by PR #9799 and PR #9783, which are independent of each other and can be merged in any order.  This PR, PR #9768, must be merged first.


### How to test?
As an instructor, move a student to the Null section and make sure you untick "Manual." Then, as that student, go to the course's page. On the No Access page you get, click the join button. Make sure that before you dropped the student, the student had done some sort of activity in the course within the past three days.


### Images
![image](https://github.com/Submitty/Submitty/assets/51209504/16e8f53f-62d0-4ccb-b629-b913f726951f)
and then,
![image](https://github.com/Submitty/Submitty/assets/51209504/2cd9e95e-e309-4a89-891b-6b7c7464b2bc)
